CRM/Eventinvitation/Form/Download.php

```diff
@@ -20,140 +22,168 @@
  */
 class CRM_Eventinvitation_Form_Download extends CRM_Core_Form {
 
-    /** @var string the tmp folder holding the PDFs */
-    public $tmp_folder;
-
-    /** @var string the URL to return to */
-    public $return_url;
-
-    public function buildQuickForm()
-    {
-        $this->tmp_folder = CRM_Utils_Request::retrieve('tmp_folder', 'String', $this);
-        $this->return_url = CRM_Utils_Request::retrieve('return_url', 'String', $this);
-
-        $this->setTitle(E::ts("Your Invitation PDFs are ready for download."));
-        $this->addButtons(
+  /**
+   * @var string the tmp folder holding the PDFs */
+  public string $tmp_folder = '';
+
+  /**
+   * @var string the URL to return to */
+  public string $return_url = '';
+
+  /**
+   * @throws CRM_Core_Exception
+   */
+  public function buildQuickForm():void {
+    /** @var string|null $tmpFolder */
+    $tmpFolder = CRM_Utils_Request::retrieve('tmp_folder', 'String', $this);
+    $this->tmp_folder = (string) $tmpFolder;
+
+    /** @var string|null $returnUrl */
+    $returnUrl = CRM_Utils_Request::retrieve('return_url', 'String', $this);
+    $this->return_url = (string) $returnUrl;
+
+    $this->setTitle(E::ts('Your Invitation PDFs are ready for download.'));
+    $this->addButtons(
+        [
             [
-                [
-                    'type' => 'submit',
-                    'name' => E::ts('Download'),
-                    'icon' => 'fa-download',
-                    'isDefault' => true,
-                ],
-                [
-                    'type' => 'done',
-                    'name' => E::ts('Back to Search'),
-                    'isDefault' => false,
-                ],
-            ]
-        );
-
-        parent::buildQuickForm();
-    }
+              'type' => 'submit',
+              'name' => E::ts('Download'),
+              'icon' => 'fa-download',
+              'isDefault' => TRUE,
+            ],
+            [
+              'type' => 'done',
+              'name' => E::ts('Back to Search'),
+              'isDefault' => FALSE,
+            ],
+        ]
+    );
+
+    parent::buildQuickForm();
+  }
+
+  /**
+   * @throws IllegalPathException
+   */
+  public function postProcess():void {
+    // this means somebody clicked download
+    $vars = $this->exportValues();
+    if (isset($vars['_qf_Download_submit'])) {
+      // verify folder
+      if (preg_match('#/eventinvitation_pdf_generator_\w+$#', $this->tmp_folder) === FALSE) {
+        throw new IllegalPathException('Illegal path!');
+      }
+
+      // download PDFs
+      // todo: verify folder
+      try {
+        // create ZIP file
+        $filename = $this->tmp_folder . DIRECTORY_SEPARATOR . 'all.zip';
+
+        // first: try with command line tool to avoid memory issues
+        $has_error = FALSE;
+        try {
+          // TODO: nothing here should be able to throw an exception.
```

Comment:
Let's just remove the `try`…`catch` then? This might just give the impression of error handling when there actually is none.